pkgs/tools/compression/zfp/default.nix

```diff
@@ -0,0 +1,64 @@
+{ cmake, cudatoolkit, fixDarwinDylibNames, fetchFromGitHub, gfortran, lib, llvmPackages, pythonPackages, stdenv, targetPlatform
+, enableCfp ? true
+, enableCuda ? builtins.elem targetPlatform.system cudatoolkit.meta.platforms
```

Comment:
Most other cuda packages don't enable it automatically because cuda is very big.

Reply:
Is enableCuda good or is there another mechanism for packages to check whether the desire is to build with CUDA support?

Reply:
There is no other great way except config.something which is also not discoverable.